### PR TITLE
Feat/ls headers

### LINKED ISSUE #893: ls headers should be a flag
`ipfs ls` recently learned to show headers:

```
> ipfs ls QmVQ1XvYGF19X4eJqz1s7FJYJqAxFC4oqh3vWJJEXn66cp
Hash                                           Size   Name
QmajFHHivh25Qb2cNbnnnEeUe1gDLHX9ta7hs2XKX1vazb 139795 cat.jpg
QmNtpA5TBNqHrKf3cLQ1AiUKXiE4JmUodbG5gXrajg8wdv 277    test/
QmWNj1pTSjbauDHpdyg5HQ26vYcNWnubg1JehmwAE9NnU9 862825 tree-in-cosmos.jpg
```
Which is great! :)
but it broke people's scripts :( -- and i think it should be opt-in.

This is a format change, and those should be done carefully, as they are liable to break other things or introduce subtle problems for users or programs. (I CRed this and didnt notice this, so it's my fault really.)

Let's:
- [ ] make it only be turned on with a flag (i suggest `-h, --header` or something similar)
- [ ] add somewhere to some dev doc that PRs that change format should be clearly indicated and require a LGTM from the format maintainer (@jbenet for the moment)

### PR DESCRIPTION
**Pull request by cryptix**
Inspired by #372, this fixes #893 .

### CODE REVIEW

File: core/commands/ls.go [hunk 1]
```diff
@@ -87,25 +98,31 @@ it contains, with the following format:
...
+		res.SetOutput(&LsOutput{
+			Objects: output,
+			Headers: headers,
+		})
 	},
 	Marshalers: cmds.MarshalerMap{
 		cmds.Text: func(res cmds.Response) (io.Reader, error) {
-			output := res.Output().(*LsOutput).Objects
+
+			output := res.Output().(*LsOutput)
```
Comment (whyrusleeping) [inline core/commands/ls.go]: you can get the request object and the options from the response here. Not sure if thats the *right* way to do it, but its what ive been doing.

Comment (jbenet) [inline core/commands/ls.go]: yeah that's the right way.

Comment (jbenet) [inline core/commands/ls.go]: so,

```go
headers, _, err := res.Request().Option("headers").Bool()
```
here

Comment (cryptix) [inline core/commands/ls.go]: ah okay, will do.

Comment (jbenet) [inline core/commands/ls.go]: Addressed in 79f2792

**Comment by jbenet** [comment]
@cryptix just fixed it! dont worry :) :+1:  thanks for this PR!

**Comment by jbenet** [comment]
i'll squash (to make sure every commit passes) when these tests pass.

**Pull request merged**